src/chain/sync.ts

```diff
@@ -93,7 +93,11 @@ export default class ChainSync {
 		return ChainSync.instance
 	}
 
-	public async start() {
+	public async getCurrentBlock(): Promise<number> {
+		return this.provider.getBlockNumber()
+	}
```

Comment:
That's another one of those that I waffle on.   By returning the non-awaited promise, we allow the caller to decide if the value is needed immediately.   If we pre-await the return, then we've removed that flexibility and can actually hide the fact that there may be missing "awaits", unless TypeScript actually warns about such things?

Having come from a heavily, explicitly, multi-threaded background, it disturbs me to see so many async functions that simply await everything internally such that they aren't really asynchronous, but are only flagged as async because it allows the compiler/interpreter to generate the proper under-the-covers constructs.

So, while I agree we should probably have a standard, my vote would go on returning un-awaited promises and leave the decision up to the caller.   Unless, of course, the function itself has a need to process the awaited value.

On a similar note, I'm guilty of flagging functions as being async that aren't really, and I need to get out of that habit.   If a function does all of its processing synchronously, and doesn't invoke anything async, then it shouldn't be flagged as async to start with.   Kind of like your "const change to let when necessary" approach that I like.

Reply:
Per our discussion, in discord, consumers of `getCurrentBlock` are expecting a value, and there doesn't seem to be a reason by which these calls would be useful in a parallel situation (by virtue of `current` block), so I would be in favour of the `async` / `await` syntax.